Assets/CohortUnityClient/Scripts/CHSession.cs

```diff
@@ -115,8 +140,181 @@ public class CHSession : MonoBehaviour {
 
     private bool socketConnectionActive = false;
 
+    private string jwtToken = "";
+    private UnityWebRequest cohortUpdateEventRequest;
+
+    private List<CHEpisode> episodesArray;
+
+    private string episodeJson;
+    private Boolean successfulServerRequest;
+
+    [Serializable]
+    public class JwtToken
+    {
+      public string jwt;
+    }
+
+
+    [Serializable]
+    public class CHEpisode
+    {
+      public int episodeNumber { get; set; }
+      public string label { get; set; }
+      public List<CHMessage> cues { get; set; }
+    }
+
+
+
+    string serverToEventUrl(string url) {
+      //this checks for any of the following words/letter sequences, but I'm not sure how to use /(localhost|.local|192.168.)/mi in this context
+      string urlInput = "(localhost|.local|192.168.)"; 
+      string cohortUpdatedEventURL;
+
+      // Instantiate the regular expression objects.
+      Regex compareUrl = new Regex(urlInput, RegexOptions.IgnoreCase);
+
+      // Match the regular expression pattern against our URL string.
+      Match matchUrl = compareUrl.Match(url);
+
+      //if match occurs add port number
+      if (matchUrl.Length > 0)
+      {
+        cohortUpdatedEventURL = url + ":" + httpPort + "/api/v2";
+      }
+      else
+      {
+        cohortUpdatedEventURL = url + "/api/v2";
+      }
+
+      Debug.Log(cohortUpdatedEventURL);
+      return cohortUpdatedEventURL;
+
+    }
+      
+    void OnValidate(){
+      //first check if we need to authenticate
+      if (jwtToken == ""){
+        Credentials userCredentials = new Credentials();
+        userCredentials.username = username;
+        userCredentials.password = password;
+        string loginJson = JsonMapper.ToJson(userCredentials);
+
+        //
+        StartCoroutine(authenticationRequest(serverToEventUrl(serverURL) + "/login?sendToken=true", loginJson));
+
+      } else {
+        
+          successfulServerRequest = false;
+          Debug.Log("OnValidate");
+          string cuesJson = jsonFromCues();
+        //uncomment to verify Json getting sent
+        //Debug.Log(jsonFromCues());
+
+        updateRemoteInfo(cuesJson);
+        }
+    }
+
+    IEnumerator authenticationRequest(string uri, string json)
+    {
+      using (UnityWebRequest cohortLoginRequest = UnityWebRequest.Put(uri, json))
+      {
+        cohortLoginRequest.SetRequestHeader("Content-Type", "application/json");
+        cohortLoginRequest.method = "POST";
+        // Request and wait for the desired page.
+        yield return cohortLoginRequest.SendWebRequest();
+
+        if (cohortLoginRequest.isNetworkError){
+          Debug.Log(" Error: " + cohortLoginRequest.error);
```

Comment:
Maybe add an 'if we're in the editor' > extra line to console telling the user what, in Unity, to try to do (i.e. specific directions to complement the server's info message).